dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/validation/DefaultValidationService.java

```diff
@@ -370,9 +369,7 @@ private void addRulesToContext( Map<PeriodType, PeriodTypeExtended> periodTypeXM
         {
             PeriodTypeExtended periodX = periodTypeXMap.get( rule.getPeriodType() );
 
-            if ( periodX == null
```

Comment:
@larshelge the only consideration is, what happens if the rule is invalid and an exception is thrown?  Is it caught and reported properly by the API or does it fails with a 500 error?

Reply:
Tested now and can see that the validation engine behaves well, by ignoring and logging invalid rules/expressions.